src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -24,15 +24,16 @@
 import org.jsoup.safety.Whitelist;
 import java.lang.String;
 
-
+// Try typing "/chats" or "/chats?userid=Elijah" or "/chats?chatid=someid"
+// to see how the query parameters work for the servlet
 @WebServlet(urlPatterns = {"/chats/*"})
```

Comment:
This looks inconsistent. The urlPattern here is `/chats/*` with a second `/` but the example doesn't have it. What's the intended API path?

Reply:
Thats just how it works. To clarify if you type "/chats" and add "?userid=abcxyz" to the end of it, the urlpattern ignores the final "/" in "/chats/*" from all indications.

Reply:
In that case, you actually don't need the `urlPatterns` property here. The `/*` piece of the path is not even being used in your code as far as I can tell. You should be able to follow the pattern found in your [portfolio](https://github.com/google/software-product-sprint/blob/main/portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java#L10).

Otherwise it looks confusing that the servlet seems to handle additional subpaths under `/chat/` when in reality it doesn't. It's also confusing for documentation to provide examples but not examples for, based on this line, the supposed support of additional subpaths.

Reply:
Ok makes sense